metals/src/main/scala/scala/meta/internal/builds/MavenBuildTool.scala

```diff
@@ -0,0 +1,63 @@
+package scala.meta.internal.builds
+
+import scala.meta.internal.metals.{MetalsServerConfig, UserConfiguration}
+import scala.meta.io.AbsolutePath
+import scala.tools.nsc.Properties
+import scala.meta.internal.metals.MetalsEnrichments._
+
+object MavenBuildTool extends BuildTool {
+
+  private lazy val embeddedMavenLauncher: AbsolutePath = {
```

Comment:
What happens if we turn `MavenBuildTool` into a `class` instead of `object`? Same for `GradleBuildTool`. I'm a concerned about doing I/O on lazy vals for global objects.

Reply:
Right, turned into classes, that wasn't an issue.